src/c/mod.rs

```diff
@@ -34,6 +34,29 @@ use crate::constants;
 use crate::types::{Account, Provider, Subscription};
 use crate::utils::watch_tx;
 
+fn client_clauses_validator(
+    client: *mut ToriiClient,
+    clauses: *const EntityKeysClause,
```

Comment:
I think we could use NonNull ptrs to not have to do that type of validation on our side. https://doc.rust-lang.org/std/ptr/struct.NonNull.html just not sure if its gonna work with cbindgen. Worth giving it a try tho

Reply:
It seems that the bindgen works fine, but it still crashes when getting a null pointer, an alternative may be using `Option<NonNull<EntityKeysClause>>`, and then we check it with `Some()`, but I think going for the raw pointers may be more straightforward.

Reply:
ah ok yeah then maybe not worth the hassle. just verifying `.is_null()` is more straightforward.